Website/src/rendering/src/components/FullPageSearch/OrderCloudFullPageSearch.tsx

```diff
@@ -0,0 +1,139 @@
+  SearchResultsSortChangedActionPayload,
+  SearchResultsKeyphraseChangedActionPayload,
+} from '@sitecore-discover/widgets';
+import { OcProductListOptions, listProducts } from '../../redux/ocProductList';
+import { clone, omit } from 'lodash';
+import { useAppDispatch } from '../../redux/store';
+import { getCategoryIdByUrlPath } from '../../helpers/CategoriesDataHelper';
+
+const OrderCloudFullPageSearch = (): JSX.Element => {
+  const dispatch = useAppDispatch();
+  const productListState = useOcProductList() as unknown as FullPageSearchResultsProps;
+  const [rfkid, setRfkid] = useState<string>('');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [facetFilters, setFacetFilters] = useState<any>({});
+  const router = useRouter();
+
+  const getCategoryIdFromUrl = (): string => {
+    if (typeof window == 'undefined') {
+      return '';
+    }
+    return getCategoryIdByUrlPath(window.location.pathname);
+  };
+
+  const buildExistingOptions = (): OcProductListOptions => {
+    let sortBy: string[] = ['Name']; // default sort
+    if (productListState.sortType) {
+      if (productListState.sortDirection === 'desc') {
+        sortBy = [`!${productListState.sortType}`];
+      } else {
+        sortBy = [productListState.sortType];
+      }
+    }
+    return {
+      categoryID: getCategoryIdFromUrl(),
+      depth: 'all',
+      search: productListState.keyphrase,
+      sortBy: sortBy,
+      page: productListState.page,
+      filters: facetFilters,
+    };
+  };
+
+  const onFacetClick = ({
+    checked,
+    facetType,
+    facetValue,
+  }: SearchResultsFacetClickedChangedActionPayload) => {
+    const options = buildExistingOptions();
+    const facetKey = `xp.${facetType}`;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let filters = clone(facetFilters);
+    if (checked) {
+      if (!filters[facetKey]) {
+        // facet has not been selected yet, initialize array
+        filters[facetKey] = [facetValue];
+      } else {
+        // facet has been previously selected but with a different option, so add option to array
+        filters[facetKey] = [...filters[facetKey], facetValue];
+      }
+    } else {
+      if (filters[facetKey].length === 1) {
+        // only one option selected so just remove the facet filter completely
+        filters = omit(filters, facetKey);
+      } else {
+        // more than one option is selected so remove the option from the existing array
+        filters[facetKey] = filters[facetKey].filter((option: string) => option !== facetValue);
+      }
+    }
+    setFacetFilters(filters);
+    options.filters = filters;
+    options.page = 1;
+    dispatch(listProducts(options));
+  };
+
+  const onClearFilters = () => {
+    const options = {};
+    setFacetFilters(options);
+    dispatch(listProducts(options));
+  };
+
+  const onPageNumberChange = ({ page }: SearchResultsPageNumberChangedActionPayload) => {
```

Comment:
With the OrderCloud implementation, clicking on the "View More" button does not load more results even if the total number exceeds the number of displayed results.